### PR TITLE
[SYCL] untie from cl_map_flags

### PR DESCRIPTION
**Pull request by cperkinsintel**
Define new pi_map_flags to use in lieu of cl_map_flags. 

Signed-off-by: Chris Perkins <chris.perkins@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/pi.h [hunk 1]
```diff
@@ -494,6 +494,13 @@ constexpr pi_mem_flags PI_MEM_FLAGS_HOST_PTR_USE = CL_MEM_USE_HOST_PTR;
 constexpr pi_mem_flags PI_MEM_FLAGS_HOST_PTR_COPY = CL_MEM_COPY_HOST_PTR;
 constexpr pi_mem_flags PI_MEM_FLAGS_HOST_PTR_ALLOC = CL_MEM_ALLOC_HOST_PTR;
 
+// CP
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/detail/pi.h:497]: Add a more meaningful comment

File: sycl/include/CL/sycl/detail/pi.h [hunk 2]
```diff
@@ -1378,9 +1385,9 @@ piEnqueueMemImageFill(pi_queue command_queue, pi_mem image,
 
 __SYCL_EXPORT pi_result piEnqueueMemBufferMap(
     pi_queue command_queue, pi_mem buffer, pi_bool blocking_map,
-    cl_map_flags map_flags, // TODO: untie from OpenCL
-    size_t offset, size_t size, pi_uint32 num_events_in_wait_list,
-    const pi_event *event_wait_list, pi_event *event, void **ret_map);
+    pi_mem_flags map_flags, size_t offset, size_t size,
```
Comment (smaslov-intel) [inline sycl/include/CL/sycl/detail/pi.h:1388]:
```suggestion
    pi_map_flags map_flags, size_t offset, size_t size,
```

**Review by smaslov-intel — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/detail/pi.h (hunk 2 above)
Comment (cperkinsintel) [reply sycl/include/CL/sycl/detail/pi.h]: d'oh!  thanks.

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 3]
```diff
@@ -4039,7 +4039,7 @@ pi_result cuda_piEnqueueMemImageFill(pi_queue command_queue, pi_mem image,
 ///
 pi_result cuda_piEnqueueMemBufferMap(pi_queue command_queue, pi_mem buffer,
                                      pi_bool blocking_map,
-                                     cl_map_flags map_flags, size_t offset,
+                                     pi_mem_flags map_flags, size_t offset,
```
Comment (smaslov-intel) [inline sycl/plugins/cuda/pi_cuda.cpp:4042]:
```suggestion
                                     pi_map_flags map_flags, size_t offset,
```

Comment (smaslov-intel) [reply sycl/plugins/cuda/pi_cuda.cpp:4042]: And other places too

**Review by smaslov-intel — CHANGES_REQUESTED** [review]

File: sycl/plugins/cuda/pi_cuda.cpp (hunk 3 above)
Comment (cperkinsintel) [reply sycl/plugins/cuda/pi_cuda.cpp]: Thanks again. They were all wrong. I went through and audited the changes and re-fixed them.

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**